### PR TITLE
usersignup phone verification api changes

### PR DESCRIPTION
**Pull request by tinakurian**
## Description
Adding annotations, labels and spec change for phone verification: 

Added constant for VerificationCode annotation name
Added constant for VerificationTimestamp annotation name
Added constant for VerificationCounter annotation name
Added constant for VerificationAttempts annotation name
Added constant for VerificationExpiry annotation name
Added constant for PhoneNumber label name
Added optional VerificationRequired boolean attribute to UserSignupSpec

Jira: https://issues.redhat.com/browse/CRT-759

## Checks
1. Have you run `make generate` target? **[yes/no]**
Yes
2. Does `make generate` change anything in other projects (host-operator, member-operator, toolchain-common)? **[yes/no]** 
Yes

3. In case other projects are changed, please provides PR links.
    - host-operator: https://github.com/codeready-toolchain/host-operator/pull/260

### CODE REVIEW

File: pkg/apis/toolchain/v1alpha1/usersignup_types.go [hunk 1]
```diff
@@ -14,9 +14,21 @@ const (
...
+	//UserSignupVerificationCodeAnnotationKey is used for the usersignup verification code annotation key
+	UserSignupVerificationCodeAnnotationKey = LabelKeyPrefix + "verification-code"
+	//UserSignupVerificationTimestamp is used for the usersignup verification timestamp annotation key
+	UserSignupVerificationTimestamp = LabelKeyPrefix + "verification-timestamp"
+	//UserSignupVerificationCounter is used for the usersignup verification counter annotation key
+	UserSignupVerificationCounter = LabelKeyPrefix + "verification-counter"
+	//UserVerificationAttempts is used for the usersignup verification attempts annotation key
+	UserVerificationAttempts = LabelKeyPrefix + "verification-attempts"
+	//UserVerficationExpiry is used for the usersignup verification expiry annotation key
+	UserVerficationExpiry = LabelKeyPrefix + "verification-expiry"
```
Comment (alexeykazakov) [inline pkg/apis/toolchain/v1alpha1/usersignup_types.go:26]:
```suggestion
	//UserSignupVerificationTimestampAnnotationKey is used for the usersignup verification timestamp annotation key
	UserSignupVerificationTimestampAnnotationKey = LabelKeyPrefix + "verification-timestamp"
	//UserSignupVerificationCounterAnnotationKey is used for the usersignup verification counter annotation key
	UserSignupVerificationCounterAnnotationKey = LabelKeyPrefix + "verification-counter"
	//UserVerificationAttemptsAnnotationKey is used for the usersignup verification attempts annotation key
	UserVerificationAttemptsAnnotationKey = LabelKeyPrefix + "verification-attempts"
	//UserVerficationExpiryAnnotationKey is used for the usersignup verification expiry annotation key
	UserVerficationExpiryAnnotationKey = LabelKeyPrefix + "verification-expiry"
```

File: pkg/apis/toolchain/v1alpha1/usersignup_types.go [hunk 2]
```diff
@@ -74,6 +86,10 @@ type UserSignupSpec struct {
 	// The user's company name, obtained from the identity provider.
 	// +optional
 	Company string `json:"company,omitempty"`
+
+	// VerificationRequired is used to determine if a user requires phone verification
+	// +optional
```
Comment (alexeykazakov) [inline pkg/apis/toolchain/v1alpha1/usersignup_types.go:91]:
```suggestion
	// VerificationRequired is used to determine if a user requires phone verification.
	// The user should not be provisioned if VerificationRequired is set to true.
	// VerificationRequired is set to false when the user is ether exempt from phone verification or has already successfully passed the verification.
	// Default value is false.
	// +optional
```

File: pkg/apis/toolchain/v1alpha1/usersignup_types.go (hunk 1 above)
Comment (tinakurian) [reply pkg/apis/toolchain/v1alpha1/usersignup_types.go]: fixed :facepalm: 3afa3a7892cb3bd2f616018f25a62edb50c81210

File: pkg/apis/toolchain/v1alpha1/usersignup_types.go (hunk 2 above)
Comment (tinakurian) [reply pkg/apis/toolchain/v1alpha1/usersignup_types.go]: updated as part of: 3afa3a7892cb3bd2f616018f25a62edb50c81210

**Pull request merged**